src/Codegen/Codegen.hs

```diff
@@ -809,12 +811,31 @@ genPrim I.Par procs _ = do
         return (ret, fnStms ++ argStms, appStms)
       apply (e, _) = do
         fail $ "Cannot compile par with non-application expression: " ++ show e
+      -- given a list of par return vals and their types, wrap the return vals in a tuple
+      genParRetVal :: [C.Exp] -> [I.Type] -> GenFn (C.Exp, [C.BlockItem])
+      genParRetVal [] _ = fail "par should have 2 or more return values"
```

Comment:
nit: You can just put this as a `_` wildcard case after the `rets@(ret0:ret1:_)` case, to consolidate them